tests/network_rules_update_test.go

```diff
@@ -0,0 +1,218 @@
+	require.NotEqual(0, originalRules.Economy.MinBaseFee.Int64(), "MinBaseFee should be filled")
+
+	newMinBaseFee := 1e3 * originalRules.Economy.MinBaseFee.Int64()
+	updateRequest := rulesType{}
+	updateRequest.Economy.MinBaseFee = new(big.Int).SetInt64(newMinBaseFee)
+
+	// Update network rules
+	updateNetworkRules(t, net, updateRequest)
+
+	// Network rule should not change - it must be an epoch bound
+	var updatedRules rulesType
+	err = client.Client().Call(&updatedRules, "eth_getRules", "latest")
+	require.NoError(err)
+
+	require.Equal(originalRules.Economy.MinBaseFee, updatedRules.Economy.MinBaseFee,
+		"Network rules should not change - it must be an epoch bound")
+
+	// produce a block to make sure the rule is not applied
+	_, err = net.EndowAccount(common.Address{}, big.NewInt(1))
+	require.NoError(err)
+
+	blockBefore, err := client.BlockByNumber(t.Context(), nil)
+	require.NoError(err)
+
+	require.Less(blockBefore.BaseFee().Int64(), newMinBaseFee, "BaseFee should not reflect new MinBaseFee")
+
+	// apply epoch change
+	advanceEpochAndWaitForBlocks(t, net)
+
+	// rule should be effective
+	err = client.Client().Call(&updatedRules, "eth_getRules", "latest")
+	require.NoError(err)
+
+	require.Equal(newMinBaseFee, updatedRules.Economy.MinBaseFee.Int64(),
+		"Network rules should become effective after epoch change")
+
+	blockAfter, err := client.BlockByNumber(t.Context(), nil)
+	require.NoError(err)
+
+	require.GreaterOrEqual(blockAfter.BaseFee().Int64(), newMinBaseFee, "BaseFee should reflect new MinBaseFee")
+}
+
+func TestNetworkRule_Update_Restart_Recovers_Original_Value(t *testing.T) {
+	require := require.New(t)
+	net := StartIntegrationTestNetWithFakeGenesis(t)
+
+	client, err := net.GetClient()
+	require.NoError(err)
+	defer client.Close()
+
+	type rulesType struct {
+		Economy struct {
+			MinBaseFee *big.Int
+		}
+	}
+
+	var originalRules rulesType
+	err = client.Client().Call(&originalRules, "eth_getRules", "latest")
+	require.NoError(err)
+	require.NotEqual(0, originalRules.Economy.MinBaseFee.Int64(), "MinBaseFee should be filled")
+
+	newMinBaseFee := 1e3 * originalRules.Economy.MinBaseFee.Int64()
+	updateRequest := rulesType{}
+	updateRequest.Economy.MinBaseFee = new(big.Int).SetInt64(newMinBaseFee)
+
+	// Update network rules
+	updateNetworkRules(t, net, updateRequest)
+
+	// Restart the network, since the rules happened withing a current epoch
+	// it should not be applied immediately but persisted to be applied at the end of the epoch.
+	err = net.RestartWithExportImport()
+	require.NoError(err)
+
+	client2, err := net.GetClient()
+	require.NoError(err)
+	defer client2.Close()
+
+	// Network rule should not change - it must be an epoch bound
+	var updatedRules rulesType
+	err = client2.Client().Call(&updatedRules, "eth_getRules", "latest")
+	require.NoError(err)
+
+	require.Equal(originalRules.Economy.MinBaseFee, updatedRules.Economy.MinBaseFee,
+		"Network rules should not change - it must be an epoch bound")
+
```

Comment:
Can you produce a block here, after the restart, and check that the old fee is still in use before the epoch boundary?